automation-roles/40-configure-infra/openshift-monitoring/tasks/enable-openshift-monitoring.yml

```diff
@@ -1,62 +1,51 @@
+
+- name: Create cluster monitor configuration
   shell: |
-    oc get cm -n openshift-monitoring --no-headers | grep cluster-monitoring-config | wc -l
-  register: _cluster_monitor_config_exists
-    
-- when: _cluster_monitor_config_exists.stdout == "0"
-  block:
-
-    - set_fact:
-        _oc_monitoring_persist_sc: "{{ _p_openshift_monitoring.pvc_storage_class | default ('') }}"
-        _oc_monitoring_persist_size_gb: "{{ _p_openshift_monitoring.pvc_storage_size_gb | default ('100') }}"
-        _oc_remote_rewrite_url: "{{ _p_openshift_monitoring.remote_rewrite_url | default ('') }}"
-
-    - name: Create cluster monitor configuration file
-      template:
-        src: enable-openshift-monitoring.j2
-        dest: "{{ status_dir }}/openshift/{{ current_openshift_cluster.name }}-cluster-monitor-config.yaml"
-
-    - name: Create cluster monitor configuration
-      shell: |
-        oc create -f {{ status_dir }}/openshift/{{ current_openshift_cluster.name }}-cluster-monitor-config.yaml
+    oc apply -f {{ status_dir }}/openshift/{{ current_openshift_cluster.name }}-user-workload-cluster-monitor-config.yaml
     
-    - name: Wait 15 seconds
-      pause:
-        seconds: 15
-
-    - name: Wait until openshift-monitoring prometheus pods are ready
-      shell: oc wait -n openshift-monitoring --for=condition=Ready pod -l app=prometheus --timeout=240s
-      register: _openshift_user_workload_pod
-
-    - name: Wait until pods are created in openshift-user-workload-monitoring project
-      shell: oc get po -n openshift-user-workload-monitoring 2>/dev/null | wc -l
-      register: _oc_user_workload
-      retries: 10
-      delay: 30
-      until: _oc_user_workload.stdout != "0"    
-
-    - name: Wait until openshift-user-workload-monitoring pods are ready
-      shell: oc wait -n openshift-user-workload-monitoring --for=condition=Ready pod --all --timeout=240s
-      register: _openshift_user_workload_pod_enable_persist
-
-    - when: _oc_monitoring_persist_sc != ""
-      block:
-        - name: Create user workload cluster monitor configuration file
-          template:
-            src: persist-user-workload-monitoring.j2
-            dest: "{{ status_dir }}/openshift/{{ current_openshift_cluster.name }}-user-workload-cluster-monitor-config.yaml"  
-
-        - name: Create cluster monitor configuration
-          shell: |
-            oc create -f {{ status_dir }}/openshift/{{ current_openshift_cluster.name }}-user-workload-cluster-monitor-config.yaml
-            
-        - name: Wait 15 seconds
-          pause:
-            seconds: 15
-
-        - name: Wait until openshift-user-workload-monitoring pods are restarted with the configured Storage
-          shell: oc wait -n openshift-user-workload-monitoring --for=condition=Ready pod/prometheus-user-workload-0 pod/prometheus-user-workload-1 --timeout=240s
-          register: _openshift_user_workload_pod_enable_persist
+- name: Wait 30 seconds
```

Comment:
Can you first wait for the pod to be created (oc get po -l ... | wc -l) and after that run the `oc wait`? Now, a 30s wait is done for every run.